### PR TITLE
add ruff and mypy

### PR DESCRIPTION
**Pull request by MP91**
- fix linter errors

### CODE REVIEW

**Review by dennismeister93 — DISMISSED** [review]
lgtm

File: runtime_k3d/src/runtime/runtime.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def is_runtime_installed(log_output: TextIOWrapper | int = subprocess.DEVNULL) -
         if key == "STATUS" and value == "deployed":
             return True
 
-    log_output.write(process.stdout)
+    log_output.write(process.stdout)  # type: ignore
```
Comment (BjoernAtBosch) [inline runtime_k3d/src/runtime/runtime.py]: I don't like the "code pollution" with linter comments. Also, there are too many places where this needs being added.
Is there no option to either fix it or comment it out "globally"?

Comment (MP91) [reply runtime_k3d/src/runtime/runtime.py]: I totally agree and would like to really fix this, but thought that this is out of scope of this PR. 

There are different solutions how to fix it, one would be to add a log function to the velocitas-lib, another one would be to simply force a valid logfile everywhere.

TLDR: needs more discussion

Comment (MP91) [reply runtime_k3d/src/runtime/runtime.py]: fixed

**Review by BjoernAtBosch — APPROVED** [review]
LGTM
